### ISSUE TITLE
[Bug] Add custom search engine accessibility updates

### ISSUE THREAD

**Issue by sblatz** [mozilla-mobile/fenix#7967]
Let's update the "add search engine" screen:

1) Make the first focus box say "Custom search engine details" (confirm string)
2) First swipe is fine (Says "name edit box"
3) Second swipe should say "Search string to use... Replace query with %s" etc. which is the text displayed below the search string to use edit box.
4) Learn More should say "Learn more Link" (this should be a link object) Chat with @eeejay about this.

**Comment by ValentinTimisica** [comment]
I investigated the 4th point and found that there is no Type announcement for "link", as also discussed here (https://stackoverflow.com/questions/43188627/android-talkback-element-type-announcement). So our solutions for this would be:
- Use a spannableString with a clickableSpan for the textView to be identified as a link and set its contentDescription with the string "Learn more Link". This would allow it to be read by accessibility services as "Learn more link" as @sblatz suggested.
- Use the same spannableString but without contentDescription case in which on some devices this will be announced as "Learn more, 1 link", on other devices this will be announced as "Learn more" and the users will be presented with a "Local Context Menu" from where they could see, and open the link and probably other implementations on other devices.

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on the latest Nightly from 2/24 with Samsung Galaxy Note 8 (Android 9), and Sony Xperia Z5 Premium (Android 7.1.1).

Using Accessibility - Voice assistant, the "Add search engine" page is read:
- first swipe: "Name edit box"
- second swipe: "Search string to use edit box for replace query with %s. Example: https://www.google.com/searchq=%s", which is the text displayed below the search string to use the edit box,
- third swipe: "Learn more Link".
